tests/validators/test_model.py

```diff
@@ -1164,3 +1164,166 @@ def f(values_or_values_and_fields_set: Any, *args: Any) -> Any:
     assert m.b == 2
     assert m.__pydantic_fields_set__ == {'a', 'b'}
     assert calls == [call1, call2]
+
+
+def test_custom_init():
```

Comment:
maybe put these in a separate file?

Reply:
I moved them to `validators/model_init`. I think it makes sense to still be under validators since this is so closely related to validator behavior

Reply:
.